packages/elements/src/react/sign-in/choose-strategy.tsx

```diff
@@ -0,0 +1,145 @@
+'use client';
+
+import type { SignInFactor, SignInFirstFactor, SignInStrategy as TSignInStrategy } from '@clerk/types';
+import { Slot } from '@radix-ui/react-slot';
+import { useSelector } from '@xstate/react';
+import * as React from 'react';
+import type { ActorRefFrom } from 'xstate';
+
+import type { TSignInFirstFactorMachine } from '~/internals/machines/sign-in/machines';
+import { SignInRouterSystemId } from '~/internals/machines/sign-in/types';
+
+import { useActiveTags } from '../hooks';
+import { ActiveTagsMode } from '../hooks/use-active-tags.hook';
+import { SignInRouterCtx } from './context';
+
+// --------------------------------- HELPERS ---------------------------------
+
+const localStrategies: Set<TSignInStrategy> = new Set(['email_code', 'password', 'phone_code', 'email_link']);
+const resetPasswordStrategies: Set<TSignInStrategy> = new Set([
+  'reset_password_phone_code',
+  'reset_password_email_code',
+]);
+
+export function isResetPasswordStrategy(strategy: TSignInStrategy | null | undefined): boolean {
+  return !!strategy && resetPasswordStrategies.has(strategy as TSignInStrategy);
```

Comment:
Nit: I personally prefer the more verbose way of writing such logic as you did in the `factorHasLocalStrategy` function.

```suggestion
  if (!strategy) return false;
  return resetPasswordStrategies.has(strategy as TSignInStrategy);
```